### PR TITLE
[SDK-3997] Upload test results to observability server

### LINKED ISSUE #1548: Upload test results to observability server

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
Generates JUnit format results for the Node and browser tests, and uploads them to the [test observability server](https://test-observability.herokuapp.com/). Based on work started by @owenpearson in #1009. See commit messages for more details.

[Here](https://test-observability.herokuapp.com/repos/ably/ably-js/uploads/64d5db28-a7b9-4c97-a236-d83891892c61) is an example upload with some failures. (Before merging this PR, I'll delete all of the existing ably-js uploads so as to not skew the results with these example failures.)

Note that the observability server UI doesn't currently display a link to the individual matrix job that caused the upload. This is addressed by the following open PRs:

- https://github.com/ably/test-observability-action/pull/20
- https://github.com/ably/test-observability/pull/76

Resolves #1548.

### CODE REVIEW

**Comment by lawrence-forooghian** [comment]
Sorry, just realised there's still a `TODO` here; converting to draft for now.

**Comment by lawrence-forooghian** [comment]
Sorry about that — ready for review now.

**Comment by lawrence-forooghian** [comment]
I've pushed 0f0dbb314e67956e3b83f0d49256b567bdcd14b1 to fix a couple of mistakes. Will squash it once the PR is approved; I don't want to rebase it at the moment since I've got a couple of other branches that are based on an earlier commit in this PR and I don't want to sever that connection right now.

File: webpack.config.js
Comment (VeskeR) [inline webpack.config.js:230]:
```suggestion
 *   1. This package is designed for Node only and hence requires polyfills of Node libraries (e.g. `stream`, `path`) — webpack takes care of this for us.
```

File: webpack.config.js
Comment (VeskeR) [inline webpack.config.js:244]: this part might be revisited when merging into v2 branch: we are using webpack 5 there, and it seems like we will be dropping support for node 12 and 14. maybe won't need `babel-loader` here at all

File: test/support/mocha_reporter.js
Comment (VeskeR) [inline test/support/mocha_reporter.js:2]: could it just be
```javascript
const MochaJUnitReporter = require('./mocha_junit_reporter/build/node');
```
?

File: test/support/runPlaywrightTests.js
Comment (VeskeR) [inline test/support/runPlaywrightTests.js:38]: remove `jUnitReport` from destructured properties, it seems it's only present in `detail` (based on playwrightSetup.js)

**Review by VeskeR — CHANGES_REQUESTED** [review]
Integration of `mocha-junit-reporter` LGTM. Happy to merge after comments are resolved.

Also, any particular reason we don't do the same for `react-hooks` tests? It seems like vitest has a [built-in JUnit reporter](https://vitest.dev/guide/reporters#junit-reporter) too.

File: webpack.config.js
Comment (lawrence-forooghian) [reply webpack.config.js:244]: Yep, I think we will be able to revisit this on that branch. Let's see where we are with the Node decision when we come to do the next merge into v2.

File: webpack.config.js
Comment (lawrence-forooghian) [reply webpack.config.js]: Changed, thanks.

File: test/support/mocha_reporter.js
Comment (lawrence-forooghian) [reply test/support/mocha_reporter.js]: Yep, changed, thanks!

**Comment by lawrence-forooghian** [comment]
> Also, any particular reason we don't do the same for `react-hooks` tests?

I didn't even consider it, to be honest. The normal test suite is where we've been seeing the flakiness, as far as I'm aware. Also the observability server doesn't currently support differentiating between different test suites in a given repository.

**Comment by VeskeR** [comment]
> The normal test suite is where we've been seeing the flakiness

I see now, seems reasonable. No need for `react-hooks` tests upload for now I think.

**Review by VeskeR — APPROVED** [review]

**Pull request merged**
